src/PostgREST/SchemaCache.hs

```diff
@@ -1139,6 +1142,35 @@ timezones = SQL.Statement sql HE.noParams decodeTimezones
     decodeTimezones :: HD.Result TimezoneNames
     decodeTimezones = S.fromList <$> HD.rowList (column HD.text)
 
+
+-- Find the postgis function that has the signature:
+-- st_asgeojson(record,...) returns text
+postgisFunc :: Bool -> SQL.Statement AppConfig Bool
+postgisFunc = SQL.Statement sql params decoder
+  where
+    params =
+      (map escapeIdent . toList . configDbSchemas >$< arrayParam HE.text) <>
+      (map escapeIdent . toList . configDbExtraSearchPath >$< arrayParam HE.text)
+    decoder = HD.singleRow (column HD.bool)
+    sql = encodeUtf8 [trimming|
+    SELECT
+      exists(
+        SELECT
+          1
+        FROM pg_catalog.pg_proc      AS p
+        JOIN pg_catalog.pg_depend    AS d
+          ON  d.objid   = p.oid
+          AND d.deptype = 'e'
+        JOIN pg_catalog.pg_extension AS e
+          ON e.oid   = d.refobjid
+        WHERE p.pronamespace   = ANY($$1::regnamespace[] || $$2::regnamespace[])
+          AND p.proname        = 'st_asgeojson'
+          AND e.extname        = 'postgis'
+          AND p.proargtypes[0] = 'record'::regtype
+          AND pg_get_function_result(p.oid) = 'text'
+      );
+    |]
+
```

Comment:
Simplest query that does the job for now, later we can augment this same query to detect other functions like `pg_csv` aggs or `postgrest-openapi`.